### PR TITLE
[dynamo] Graph break when faking named tensors 

### LINKED ISSUE #120644: ~53 Dynamo tests are failing with "Failed running call_function <built-in method empty of type object at 0xDEADBEEF".
21 Dynamo test are failing with "Failed running call_function <built-in method empty of type object at 0xDEADBEEF".

## Repro

`PYTORCH_TEST_WITH_DYNAMO=1 pytest test_torch.py -v -k test_deterministic_empty_cpu_uint64`

You will need to remove the skip or expectedFailure before running the repro command.
This may be deleting a file in https://github.com/pytorch/pytorch/tree/main/test/dynamo_skips or https://github.com/pytorch/pytorch/tree/main/test/dynamo_expected_failures 

## Failing tests

Here's a comprehensive list of tests that fail (as of this issue) with the above message:
<details>
<summary>Click me</summary>
```
test_torch.py::TestTorchDeviceTypeCPU.test_deterministic_empty_cpu_uint64
test_namedtensor.py::TestNamedTensor.test_as_strided
test_namedtensor.py::TestNamedTensor.test_binary_ops
test_namedtensor.py::TestNamedTensor.test_copy_transpose
test_namedtensor.py::TestNamedTensor.test_factory_edge_cases
test_namedtensor.py::TestNamedTensor.test_has_names
test_namedtensor.py::TestNamedTensor.test_info_smoke
test_namedtensor.py::TestNamedTensor.test_reduction_fns
test_namedtensor.py::TestNamedTensor.test_rename
test_namedtensor.py::TestNamedTensor.test_rename_
test_namedtensor.py::TestNamedTensor.test_rename_globber
test_namedtensor.py::TestNamedTensor.test_rename_rename_map
test_namedtensor.py::TestNamedTensor.test_select
test_namedtensor.py::TestNamedTensor.test_set_names_property
test_namedtensor.py::TestNamedTensor.test_size
test_namedtensor.py::TestNamedTensor.test_split_fns_propagates_names
test_namedtensor.py::TestNamedTensor.test_stride
test_namedtensor.py::TestNamedTensor.test_unary_propagate_names_fns
test_namedtensor.py::TestNamedTensor.test_using_seen_interned_string_doesnt_bump_refcount
test_namedtensor.py::TestNamedTensor.test_using_unseen_interned_string_bumps_refcount_permanently
test_namedtensor.py::TestNamedTensor.test_using_unseen_uninterned_string_refcounts
```
</details>

cc @ezyang @msaroufim @bdhirsh @anijain2305 @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng

### PR DESCRIPTION
**Pull request by jamesjwu**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #120906
* __->__ #120779

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler @amjames

Fixes #120644

### CODE REVIEW

File: torch/_subclasses/fake_impls.py [hunk 1]
```diff
@@ -156,6 +156,9 @@ def constructors(fake_mode, func, *args, **kwargs):
     _, new_kwargs = normalize_function(
         func, args=args, kwargs=kwargs, normalize_to_only_use_kwargs=True
     )
+    if "names" in kwargs:
+        raise UnsupportedOperatorException("torch.compile doesn't support named tensors")
```
Comment (jamesjwu) [inline torch/_subclasses/fake_impls.py:160]: TODO: should we raise the same exception in both places?

**Review by zou3519 — APPROVED** [review]
looks good, thank you

File: torch/_subclasses/fake_impls.py (hunk 1 above)
Comment (zou3519) [reply torch/_subclasses/fake_impls.py]: I'm not sure what the difference is, but it looks consistent with how other exceptions are being thrown in the respective files

**Pull request closed without merging**